app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -509,6 +554,102 @@ public void onTabReselected(TabLayout.Tab tab) {
+            if (result.getResultCode() == RESULT_OK) {
+                    Intent intent = result.getData();
+                    if (intent == null){
+                        Log.d("photo picker", "photo picker returned without an intent");
+                        return;
+                    }
+                    Uri uri = intent.getData();
+                    startCropperUri(uri);
+            }
+        });
+
+        mCardIdAndBarCodeEditorLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+            if (result.getResultCode() == RESULT_OK) {
+                Intent intent = result.getData();
+                if (intent == null){
+                    Log.d("barcode card id editor", "barcode and card id editor picker returned without an intent");
+                    return;
+                }
+                BarcodeValues barcodeValues = Utils.parseSetBarcodeActivityResult(Utils.BARCODE_SCAN, result.getResultCode(), intent, getApplicationContext());
+
+                cardId = barcodeValues.content();
+                barcodeType = barcodeValues.format();
+                barcodeId = "";
+            }
+        });
+
+        mCropperLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+            Intent intent = result.getData();
+            if (intent == null){
+                Log.d("cropper", "ucrop returned a null intent");
+                return;
+            }
+            if (result.getResultCode() == Activity.RESULT_OK) {
+                Uri debugUri = UCrop.getOutput(intent);
+                if (debugUri == null){
+                    throw new RuntimeException("ucrop returned success but not destination uri!");
+                }
+                Log.d("cropper", "ucrop produced image at " + debugUri);
+                Bitmap bitmap = BitmapFactory.decodeFile(getCacheDir() + "/" + TEMP_CROP_IMAGE_NAME);
+
+                if (bitmap != null) {
+                    bitmap = Utils.resizeBitmap(bitmap);
+                    if (requestedFrontImage()) {
+                        mFrontImageUnsaved = true;
+                        setCardImage(cardImageFront, bitmap);
+                    } else {
+                        mBackImageUnsaved = true;
+                        setCardImage(cardImageBack, bitmap);
+                    }
+                    Log.d("cropper", "mRequestedImage: " + mRequestedImage);
+                    mCropperFinishedType = mRequestedImage;
+                    hasChanged = true;
+                } else {
+                    Toast.makeText(LoyaltyCardEditActivity.this, R.string.errorReadingImage, Toast.LENGTH_LONG).show();
+                }
+            }else if(result.getResultCode() == UCrop.RESULT_ERROR){
+                Throwable e = UCrop.getError(intent);
+                if (e == null){
+                    throw new RuntimeException("ucrop returned error state but not and error!");
```

Comment:
Should be "an error", not "and error". Small typo, but if I ever get this exception for whatever reason that typo may confuse future me :P